drivers/storage/portworx/portworx.go

```diff
@@ -467,6 +467,56 @@ func (p *portworx) getCurrentMaxStorageNodesPerZone(
 	return 0, fmt.Errorf("storage disabled")
 }
 
+func (p *portworx) preflightShouldRun(toUpdate *corev1.StorageCluster) bool {
+
+	if !pxutil.IsFreshInstall(toUpdate) { // Preflight should only run freshInstall
+		return false
+	}
+
+	clusterPXver := pxutil.GetPortworxVersion(toUpdate)
+
+	pxVer30, _ := version.NewVersion("3.0")
+	if !clusterPXver.GreaterThanOrEqual(pxVer30) { // Preflight should only run if the PX version is 3.0.0 and above
+		return false
+	}
+
+	check, ok := toUpdate.Annotations[pxutil.AnnotationPreflightCheck]
+	check = strings.TrimSpace(strings.ToLower(check))
+	if ok && check == "true" { // Preflight is enabled
+		return true
```

Comment:
if `check` is `false` or `skip` , should we proceed?

Reply:
Yeah if its already set we should return true...